### PR TITLE
refactor dashboard as a component library

### PR DESCRIPTION
**Pull request by skudasov**

### CODE REVIEW

File: charts/chainlink-cluster/dashboard/cmd/deploy.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import (
+	"github.com/K-Phoen/grabana/dashboard"
+	lib "github.com/smartcontractkit/chainlink/dashboard-lib/lib"
+	core_don "github.com/smartcontractkit/chainlink/dashboard-lib/lib/core-don"
+	k8spods "github.com/smartcontractkit/chainlink/dashboard-lib/lib/k8s-pods"
+	waspdb "github.com/smartcontractkit/wasp/dashboard"
+)
+
+const (
+	DashboardName = "Chainlink Cluster (DON)"
```
Comment (Atrax1) [inline charts/chainlink-cluster/dashboard/cmd/deploy.go:12]: Do we want to hardcore dashboard name ?

File: charts/chainlink-cluster/dashboard/cmd/deploy.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+		),
+	)
+	db.Add(
+		k8spods.New(
+			k8spods.Props{
+				PrometheusDataSource: cfg.DataSources.Prometheus,
+				LokiDataSource:       cfg.DataSources.Loki,
+			},
+		),
+	)
```
Comment (Atrax1) [inline charts/chainlink-cluster/dashboard/cmd/deploy.go:38]: Could we add condition to only add it for platform = kubernetes

**Review by Atrax1 — COMMENTED** [review]
Just a few little things the rest looks good and is working

File: charts/chainlink-cluster/dashboard/cmd/deploy.go (hunk 1 above)
Comment (skudasov) [reply charts/chainlink-cluster/dashboard/cmd/deploy.go:12]: I think we should have some vars static, there is no need to have more than one dashboard for `core` right now so I'd like to be sure no one deploys another dashboard @Atrax1 😄

File: charts/chainlink-cluster/dashboard/cmd/deploy.go (hunk 2 above)
Comment (skudasov) [reply charts/chainlink-cluster/dashboard/cmd/deploy.go:38]: Good catch, will do

**Pull request merged**